From 5e274586fa3a80a59c1659aa4001b6f16471272c Mon Sep 17 00:00:00 2001
From: Pauli Nieminen <pauli.nieminen@linux.intel.com>
Date: Thu, 3 May 2012 16:52:06 +0300
Subject: Remove -ffunction-sections that breaks debug symbols

gcc documentation states that -ffunction-sections and -fdata-sections are
excepted to breakd debug symbols. The problem is specially visible in
mesa where most of mesa core appears to lose debug symbols.

But as mesa isn't only one getting hit by the problem it would be best
for development to disable the questionable optimization flag.

The linker optimization is unlikely to produce noticeable performance boost
without profile guided optimization to help making sane decissions.

Change-Id: I8a5f60cc1ae770f443ae27292d6d29796c9cea41
Signed-off-by: Pauli Nieminen <pauli.nieminen@linux.intel.com>
---
 core/combo/TARGET_linux-x86.mk |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/core/combo/TARGET_linux-x86.mk b/core/combo/TARGET_linux-x86.mk
index eacbba9..27a1562 100644
--- a/core/combo/TARGET_linux-x86.mk
+++ b/core/combo/TARGET_linux-x86.mk
@@ -93,7 +93,6 @@ TARGET_GLOBAL_CFLAGS += \
 			-D_FORTIFY_SOURCE=1 \
 			-Wstrict-aliasing=2 \
 			-fPIC -fPIE \
-			-ffunction-sections \
 			-finline-functions \
 			-finline-limit=300 \
 			-fno-inline-functions-called-once \
-- 
1.7.4.1

